models/issue_comment.go

```diff
@@ -566,6 +598,45 @@ func CreateRefComment(doer *User, repo *Repository, issue *Issue, content, commi
 	return err
 }
 
+// ClearPullPushComent clear all the push commit on issue since fore push
+func ClearPullPushComent(issue *Issue) error {
+	_, err := x.Delete(&Comment{
+		Type:    CommentTypePullPushCommit,
+		IssueID: issue.ID,
+	})
+	return err
+}
+
+// CreatePullPushComment creates a commit when push commit to a pull request.
+func CreatePullPushComment(doer *User, repo *Repository, issue *Issue, content, commitSHA, repoFullName string) error {
+	if len(commitSHA) == 0 {
+		return fmt.Errorf("cannot create reference with empty commit SHA")
+	}
+
+	// Check if same reference from same commit has already existed.
+	has, err := x.Get(&Comment{
+		Type:      CommentTypePullPushCommit,
+		IssueID:   issue.ID,
+		CommitSHA: commitSHA,
+	})
+	if err != nil {
+		return fmt.Errorf("check pull push comment: %v", err)
+	} else if has {
```

Comment:
remove the `else`? If code is reachable at this point, it always means `err == nil`, since the `err != nil` branch returns.